### ISSUE TITLE
RCS1083: false positive

### ISSUE THREAD

**Issue by TineSvete** [JosefPihrt/Roslynator#200]
Hello, I am getting what I think is false positive for RCS1083 ...

`
IEnumerable<T> items;
for (int i = 0; i < items.Count(); i++)
{
	// do something
}`

**Comment by JosefPihrt** [comment]
`items.Skip(i).Any()` is equivalent to `i < items.Count()` and it is more efficient because it iterates as less items as possible.

Your example should be rewritten to:
```csharp
IEnumerable<T> items = null;

int i = 0;
foreach (T item in items)
{
    // do something 

    i++;
}
```

The diagnostic warn you that the code is not optimal so I would keep it.

**Comment by TineSvete** [comment]
I agree, it seems I jumped to wrong conclusion a bit too soon. Sorry for the inconvenience. As far as I'm concerned, you can mark this as closed.

**Comment by dotnetjunkie** [comment]
Prevent reiterating the `IEnumerable<T>` more than once whenever possible. If you require the use of the `i` index variable, rewrite it to the following:

``` c#
foreach ((int i, T item) in items.Select((i, item) => (i, item)))
{
}
```

This uses C# 7 value tuples.

**Comment by JosefPihrt** [comment]
#315
